libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -873,14 +873,48 @@ void AP_GPS::update(void)
 #endif
 }
 
+/*
+    determines if any gps's are degraded
+*/
+uint8_t AP_GPS::update_and_count_degraded_gps(void) {
+    uint8_t count = 0;
+    for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
+        if(state[i].status < GPS_OK_FIX_3D) {
+            _degraded_gps[i] = true;
+            count++;
+        }
+    }
+    return count;
+}
+
 /*
   update primary GPS instance
  */
 void AP_GPS::update_primary(void)
 {
 #if defined(GPS_BLENDED_INSTANCE)
+    uint8_t num_failed = update_and_count_degraded_gps();
+
+    if(num_failed > 0 && GPS_MAX_RECEIVERS > 1) {
```

Comment:
nitpick: please add a space between `if` and `(`. Same goes for elsewhere throughout this PR. Thank you!